client/app/telecom/telephony/line/phone/order/telecom-telephony-line-phone-order.controller.js

```diff
@@ -289,5 +290,37 @@ angular.module("managerApp").controller("TelecomTelephonyLinePhoneOrderCtrl", fu
         return self.phone && self.order.phone && ("phone." + self.order.phone) === self.phone.brand;
     };
 
+    self.ipValidator = (function () {
+        return {
+            test: function (value) {
+                return IpAddress.isValidPublicIp4(value);
+            }
+        };
+    })();
+
+    self.detachPhone = function () {
+        self.isDetaching = true;
+        OvhApiTelephony.Line().Lexi().dissociateDevice({
+            billingAccount: $stateParams.billingAccount,
+            serviceName: $stateParams.serviceName
+        }, {
+            ipAddress: self.attachedPhoneIpAddress,
+            macAddress: self.phone.macAddress
+        }).$promise.then(function () {
+            Toast.success($translate.instant("telephony_line_phone_order_detach_device_success"));
+
+            // Cache reset
+            OvhApiTelephony.Line().Lexi().resetAllCache();
+            OvhApiTelephony.Line().Phone().Lexi().resetAllCache();
+            TelephonyMediator.resetAllCache();
+            init();
+        }).catch(function (err) {
+            Toast.error($translate.instant("telephony_line_phone_order_detach_device_error"));
```

Comment:
Please add the reason of the error, it will be more easy to debug problems with the error message from API :
```javascript
Toast.error([$translate.instant("telephony_line_phone_order_detach_device_error"), _.get(err, "data.message")].join(" "));
```